src/libraries/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs

```diff
@@ -706,6 +703,32 @@ public void TestMultipleServerBind()
             connection.Timeout = new TimeSpan(0, 3, 0);
         }
 
+#if NET
+        [ConditionalFact(nameof(LdapConfigurationExists))]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public void StartNewTlsSessionContext()
+        {
+            using (var connection = new LdapConnection("server"))
+            {
+                LdapSessionOptions options = connection.SessionOptions;
+
+                // A complete test would be to use TrustedCertificatesDirectory along with other valid options to connect.
```

Comment:
Is it possible to go a little further by setting this up and trying an existing directory to make sure things work correctly?  Maybe we don't need to actually rely on it using certificates in that directory, but just make sure we can set it and things don't fall over.

Should we add any tests that try to give this new API bad data and make sure it doesn't fail in unusual ways (like AV or creating a directory that doesn't exist, etc)?

Reply:
I did push a commit that expanded this a bit. I found that calling StartNewTlsSessionContext() + Bind() throws if TrustedCertificatesDirectory() is not set.